Platform/OpenNtfsDxe/Disc.c

```diff
@@ -0,0 +1,760 @@
+
+  ASSERT (Attr != NULL);
+  ASSERT (Mft  != NULL);
+
+  Status = InitAttr (Attr, Mft);
+  if (EFI_ERROR (Status)) {
+    return NULL;
+  }
+
+  AttrStart = FindAttr (Attr, Type);
+  if (AttrStart == NULL) {
+    FreeAttr (Attr);
+    return NULL;
+  }
+
+  if ((Attr->Flags & NTFS_AF_ALST) == 0) {
+    while (TRUE) {
+      AttrStart = FindAttr (Attr, Type);
+      if (AttrStart == NULL) {
+        break;
+      }
+
+      if ((Attr->Flags & NTFS_AF_ALST) != 0) {
+        return AttrStart;
+      }
+    }
+
+    FreeAttr (Attr);
+
+    Status = InitAttr (Attr, Mft);
+    if (EFI_ERROR (Status)) {
+      return NULL;
+    }
+
+    AttrStart = FindAttr (Attr, Type);
+    if (AttrStart == NULL) {
+      FreeAttr (Attr);
+      return NULL;
+    }
+  }
+
+  return AttrStart;
+}
+
+UINT8 *
+FindAttr (
+  IN NTFS_ATTR *Attr,
+  IN UINT32    Type
+  )
+{
+  EFI_STATUS          Status;
+  UINT8               *AttrStart;
+  ATTR_HEADER_RES     *Res;
+  ATTR_HEADER_NONRES  *NonRes;
+  ATTR_LIST_RECORD    *LRecord;
+  FILE_RECORD_HEADER  *FRecord;
+  UINT64              BufferSize;
+
+  ASSERT (Attr != NULL);
+
+  if ((Attr->Flags & NTFS_AF_ALST) != 0) {
+  retry:
+    while ((Attr->Next + sizeof (ATTR_LIST_RECORD)) <= Attr->Last) {
+      Attr->Current = Attr->Next;
+      LRecord = (ATTR_LIST_RECORD *) Attr->Current;
+
+      if (BufferSize < LRecord->RecordLength) {
+        DEBUG ((DEBUG_INFO, "NTFS: (FindAttr #0) $ATTRIBUTE_LIST is corrupted.\n"));
+        return NULL;
+      }
+
+      Attr->Next += LRecord->RecordLength;
+      BufferSize -= LRecord->RecordLength;
+      if (Attr->Next <= Attr->Current) {
+        DEBUG ((DEBUG_INFO, "NTFS: (FindAttr #1) $ATTRIBUTE_LIST is corrupted.\n"));
+        return NULL;
+      }
+
+      if ((LRecord->Type == Type) || (Type == 0)) {
+        if (Attr->Flags & NTFS_AF_MFT_FILE) {
+          // Status = DiskRead (
+          //   Attr->BaseMftRecord->File->FileSystem,
+          //   ReadUnaligned32((UINT32 *)(Attr->Current + 0x10)),
+          //   512,
+          //   Attr->ExtensionMftRecord
+          //   );
+          // if (EFI_ERROR (Status)) {
+          //   DEBUG ((DEBUG_INFO, "NTFS: Could not read first part of extension record.\n"));
+          //   return NULL;
+          // }
+          //
+          // Status = DiskRead (
+          //   Attr->BaseMftRecord->File->FileSystem,
+          //   ReadUnaligned32((UINT32 *)(Attr->Current + 0x14)),
+          //   512,
+          //   Attr->ExtensionMftRecord + 512
+          //   );
+          // if (EFI_ERROR (Status)) {
+          //   DEBUG ((DEBUG_INFO, "NTFS: Could not read second part of extension record.\n"));
+          //   return NULL;
+          // }
+          //
+          // Status = Fixup (Attr->ExtensionMftRecord, mFileRecordSize, SIGNATURE_32 ('F', 'I', 'L', 'E'));
+          // if (EFI_ERROR (Status)) {
+          //   DEBUG ((DEBUG_INFO, "NTFS: Fixup failed.\n"));
+          //   return NULL;
+          // }
```

Comment:
Similar question, about origin of commented out code, and certainty (or otherwise) that it will not be needed.